### ISSUE TITLE
Duplicate headers

### ISSUE THREAD

**Issue by Rajh** [spring-projects/spring-kafka#2335]
Hello,

Why `KafkaHeaders.EXCEPTION_CAUSE_FQCN` is equal to `KafkaHeaders.EXCEPTION_FQCN` ? Is it a  typo ?

https://github.com/spring-projects/spring-kafka/blob/c0e210c61eb78bb2f8a21f2970e3a5af56ab60f2/spring-kafka/src/main/java/org/springframework/kafka/support/KafkaHeaders.java#L252

https://github.com/spring-projects/spring-kafka/blob/c0e210c61eb78bb2f8a21f2970e3a5af56ab60f2/spring-kafka/src/main/java/org/springframework/kafka/support/KafkaHeaders.java#L246

And why having 2 sets of headers for DLT (`DeadLetterPublishingRecoverer` and `@RetryableTopic` DLT) ?
`KafkaHeaders.DLT_EXCEPTION_CAUSE_FQCN` `KafkaHeaders.EXCEPTION_CAUSE_FQCN`

**Comment by garyrussell** [comment]
Thanks; fixed. The `DLT_` headers are for "classic" (blocking) retries with the default `DeadLetterPublishingRecoverer`.

The ones without the `DLT_` are for non-blocking retries; they don't have DLT in the name because the intermediate topics are not dead letter topics, they are delayed retry topics.

There's an argument that says the final publish to the retryable topic DLT should have the DLT prefix but they don't.

Bottom line; use the DLT_ headers when not using retryable topics and the others when you are.
